shared/src/clients/whatsapp-client.class/index.ts

```diff
@@ -38,27 +48,103 @@ export default class WhatsAppClient {
+        !this.credentials.authTokenOne ||
+        !this.credentials.authTokenTwo ||
+        !this.credentials.authTokenOneExpiry ||
+        !this.credentials.authTokenTwoExpiry
+      ) {
+        throw new Error(
+          'Auth tokens are required when running WhatsApp client locally'
+        )
+      }
+      this.authTokenOneObj = {
+        token: this.credentials.authTokenOne,
+        expiry: new Date(this.credentials.authTokenOneExpiry),
+      }
+      this.authTokenTwoObj = {
+        token: this.credentials.authTokenTwo,
+        expiry: new Date(this.credentials.authTokenTwoExpiry),
+      }
+      this.isLocal = true
+    }
   }
 
-  private getCredentials(
+  private async getCredentials(
     apiClient: WhatsAppTemplateMessageToSend['apiClient']
-  ): {
+  ): Promise<{
     token: string
     url: string
-  } {
+  }> {
+    if (!this.isLocal) await this.checkAndRefreshTokens()
+    if (!this.authTokenOneObj || !this.authTokenTwoObj) {
+      throw new Error('Auth tokens not found')
+    }
+    if (this.authTokenOneObj.expiry.getTime() < new Date().getTime()) {
+      throw new Error('Auth token one has expired')
+    }
+    if (this.authTokenTwoObj.expiry.getTime() < new Date().getTime()) {
+      throw new Error('Auth token two has expired')
+    }
     switch (apiClient) {
       case WhatsAppApiClient.clientOne:
         return {
-          token: this.credentials['authTokenOne'],
+          token: this.authTokenOneObj.token,
           url: this.credentials['onPremClientOneUrl'],
         }
       case WhatsAppApiClient.clientTwo:
         return {
-          token: this.credentials['authTokenTwo'],
+          token: this.authTokenTwoObj.token,
           url: this.credentials['onPremClientTwoUrl'],
         }
     }
   }
+  private async checkAndRefreshTokens() {
+    const lessThanThreeDays = (tokenDate: Date) => {
+      const now = new Date()
+      const diff = now.getTime() - tokenDate.getTime()
+      const days = diff / (1000 * 3600 * 24)
+      return days < 3
+    }
+    const refreshAuthTokenOne =
+      !this.authTokenOneObj || lessThanThreeDays(this.authTokenOneObj.expiry)
+    const refreshAuthTokenTwo =
+      !this.authTokenTwoObj || lessThanThreeDays(this.authTokenTwoObj.expiry)
+    await Promise.all([
+      refreshAuthTokenOne && this.refreshAuthTokens('1'),
+      refreshAuthTokenTwo && this.refreshAuthTokens('2'),
+    ])
+  }
+  private async refreshAuthTokens(client: '1' | '2') {
+    const {
+      data: { users },
+    } = await this.axiosInstance.request<UsersLogin200Response>({
+      method: 'post',
+      url: AUTH_ENDPOINT,
+      baseURL:
+        client === '1'
+          ? this.credentials['onPremClientOneUrl']
+          : this.credentials['onPremClientTwoUrl'],
+      headers: {
+        Authorization: `Basic ${
+          client === '1'
+            ? this.credentials.adminCredentialsOne
+            : this.credentials.adminCredentialsTwo
+        }`,
+      },
+      data: {},
+    })
```

Comment:
TODO: log something that will be monitored and caught by Sentry/Datadog, such that if refresh fails, we can continue to work normally, but we'll be alerted and be able to do something to fix it.

This is quite unlikely to happen though.